api/app/serializers/spree/v2/storefront/country_serializer.rb

```diff
@@ -12,6 +12,12 @@ class CountrySerializer < BaseSerializer
         end
 
         has_many :states, if: proc { |_record, params| params && params[:include_states] }
+
+        has_many :checkout_zone_applicable_states,
+                 serializer: :state,
```

Comment:
@nnande let's use the full name of the serializer `::Spree::V2::Storefront::StateSerializer`